### PR TITLE
New FFT API and boost::stacktrace fix

### PR DESCRIPTION
**Pull request by HaiwangYu**

### CODE REVIEW

**Comment by lgarren** [comment]
These changes are needed for wirecell v0_18_0.

**Comment by brettviren** [comment]
FYI, they are all good from my view.

File: larwirecell/Components/CMakeLists.txt [hunk 1]
```diff
@@ -25,4 +25,5 @@ art_make_library(
     lardataobj_RecoBase
     lardataobj_Simulation
     larevt_CalibrationDBI_IOVData
+    dl
```
Comment (knoepfel) [inline larwirecell/Components/CMakeLists.txt]: Only suggestion is to replace all explicit uses of `dl` with `${CMAKE_DL_LIBS}` in your CMakeLists.txt files.

**Comment by knoepfel** [comment]
@HaiwangYu, see https://cmake.org/cmake/help/v3.22/variable/CMAKE_DL_LIBS.html.

**Comment by knoepfel** [comment]
Any update on this PR @HaiwangYu?

**Comment by knoepfel** [comment]
@HaiwangYu, have you had a chance to look at this yet?

File: larwirecell/Components/CMakeLists.txt (hunk 1 above)
Comment (HaiwangYu) [reply larwirecell/Components/CMakeLists.txt]: Sure, will update this.

**Comment by knoepfel** [comment]
trigger build

**Comment by lgarren** [comment]
Oh, right this PR cannot be built without wirecell 0.18.0

**Comment by lgarren** [comment]
@HaiwangYu  wirecell v0_18_0 is installed on larsoft cvmfs.  Would you please update the ups/product_deps in this PR to use wirecell v0_18_0 so we can run the CI?  Thanks

**Comment by HaiwangYu** [comment]
@lgarren, thanks! I think we have all three components synchronized for the IDFT breading change: `wirecell` (0.18.0), `larwirecell` ([this PR](https://github.com/LArSoft/larwirecell/pull/15)) and `dunereco` ([this PR](https://github.com/DUNE/dunereco/pull/6)). It would be great if you can include this PR in a CI test and including all of them in the next larsoft release.

**Comment by lgarren** [comment]
@HaiwangYu we need you to push the update to ups/product_deps and then we can proceed

**Comment by HaiwangYu** [comment]
@lgarren, just updated ups/product_deps. Thanks!

**Comment by lgarren** [comment]
trigger build with pull requests #15, DUNE/dunereco#6

**Comment by chissg** [comment]
trigger build with pull requests https://github.com/LArSoft/larwirecell/pull/15, https://github.com/DUNE/dunereco/pull/6

**Comment by lgarren** [comment]
trigger build

**Comment by lgarren** [comment]
trigger build with pull requests https://github.com/LArSoft/larwirecell/pull/15, https://github.com/DUNE/dunereco/pull/6

**Comment by lgarren** [comment]
trigger build with pull requests #15, DUNE/dunereco#6

The previous command not pick up the dunereco PR - trying again

**Comment by lgarren** [comment]
When triggered by hand, the dune CI is fine.
`trigger --build-delay 0 --cert /tmp/xxxx --version develop --workflow dunesw_wf --revisions "LArSoft/larwirecell#15  DUNE/dunereco#6"
`
https://dbweb0.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/14846&builds=

**Comment by gartung** [comment]
trigger build with pull requests #15, https://github.com/DUNE/dunereco/pull6

**Comment by gartung** [comment]
trigger build with pull requests #15, https://github.com/DUNE/dunereco/pull/6

**Comment by gartung** [comment]
The tests are being triggered in jenkins.
Tested with other pull request(s) LArSoft/DUNE/dunereco#6

**Comment by gartung** [comment]
The tests are being triggered in jenkins.
Tested with other pull request(s) LArSoft/DUNE/dunereco#6

**Comment by lgarren** [comment]
icarus tests complete with warnings:
https://dbweb8.fnal.gov:8443/LarCI/app/ns:ICARUS/view_builds/index?offset=0&builds=icarus_ci/3942&builds=
sbnd tests complete with warnings:
https://dbweb8.fnal.gov:8443/LarCI/app/ns:SBND/view_builds/index?offset=0&builds=sbnd_ci/8898&builds=

**Comment by gartung** [comment]
The tests are being triggered in jenkins.
Tested with other pull request(s) DUNE/dunereco#6

**Comment by gartung** [comment]
The tests are being triggered in jenkins.
Tested with other pull request(s) DUNE/dunereco#6

**Comment by lgarren** [comment]
uboone build failure:
https://dbweb8.fnal.gov:8443/LarCI/app/ns:uBooNE/view_builds/index?offset=0&builds=uboone_ci/7266&builds=

**Comment by gartung** [comment]
The tests are being triggered in jenkins.
Tested with other pull request(s) DUNE/dunereco#6

**Comment by gartung** [comment]
The tests are being triggered in jenkins.
Tested with other pull request(s) DUNE/dunereco#6

**Comment by gartung** [comment]
trigger build with pull requests https://github.com/LArSoft/larwirecell/pull/15, https://github.com/DUNE/dunereco/pull/6

**Comment by lgarren** [comment]
@HaiwangYu the uboone unit tests are failing.  This is after I added ${CMAKE_DL_LIBS} to ubsim/DetSim/MicroBooNEWireCell/CMakeLists.txt.   Would you please make a feature branch for uboone?

**Comment by HaiwangYu** [comment]
@lgarren, you mean this error in [this file](https://dbweb8.fnal.gov:8443/LarCI/app/ns:uboone/storage/docs/2022/03/23/builduBooNE%23TTFL02g.log)?
```
2483: ubsim/ubsim/DetSim/MicroBooNEWireCell/CMakeFiles/WireCellubsim.dir/ReweightedDepoTransform.cxx.o: In function `boost::stacktrace::frame::name[abi:cxx11]() const':
2484: /cvmfs/larsoft.opensciencegrid.org/products/boost/v1_75_0/Linux64bit+3.10-2.17-e20-prof/include/boost/stacktrace/detail/frame_unwind.ipp:94: undefined reference to `dladdr'
2485: ubsim/ubsim/DetSim/MicroBooNEWireCell/CMakeFiles/WireCellubsim.dir/ReweightedDepoTransform.cxx.o: In function `boost::stacktrace::detail::location_from_symbol::location_from_symbol(void const*)':
2486: /cvmfs/larsoft.opensciencegrid.org/products/boost/v1_75_0/Linux64bit+3.10-2.17-e20-prof/include/boost/stacktrace/detail/location_from_symbol.hpp:31: undefined reference to `dladdr'
2487: collect2: error: ld returned 1 exit status
```

**Comment by lgarren** [comment]
@HaiwangYu see
https://dbweb8.fnal.gov:8443/LarCI/app/ns:uBooNE/view_builds/index?offset=0&builds=uboone_ci/7268&builds=

**Comment by lgarren** [comment]
@HaiwangYu I fixed the build problem with ubsim feature/lg_wirecell_0_18_0

**Comment by HaiwangYu** [comment]
@lgarren, I see. You mean here?
https://dbweb8.fnal.gov:8443/LarCI/app/ns:uboone/storage/docs/2022/03/23/unit_testuBooNE%23SKDNXbs.log

**Comment by lgarren** [comment]
> @lgarren, I see. You mean here? https://dbweb8.fnal.gov:8443/LarCI/app/ns:uboone/storage/docs/2022/03/23/unit_testuBooNE%23SKDNXbs.log

yes

**Comment by lgarren** [comment]
@HaiwangYu - have you had a chance to look at the problem?  More details on the ticket.
https://cdcvs.fnal.gov/redmine/issues/26582

**Comment by lgarren** [comment]
trigger build with pull requests #15, DUNE/dunereco#6, uboonedata@feature/yuhw_idft

**Comment by lgarren** [comment]
approve

After email exchange.

**Pull request merged**